### ISSUE TITLE
Improve training stablity

### ISSUE THREAD

**Issue by purzelrakete** [feldberlin/wavenet#11]
## What

After a couple of days of training on Maestro, the model collapsed. See the [run here](https://wandb.ai/purzelrakete/feldberlin-wavenet/runs/32sr4rxu?workspace=user-purzelrakete). Gradients are going up to values of 20k, and in some cases biases have values of 1000.

To improve the stability, initialisation will have to be looked at carefully.

## Hypothesis

Exploding gradients prevent further training on large datasets. This is due to incorrect initialisation of the network. Introducing the correct normalisation will resolve this problem, and a complete run on Maestro will complete without issue. Convergence of the network should also speed up.

## Results

Write up the results of your experiment once it has completed and has been analysed. Include links to the treatment run, and also to the baseline if appropriate.

## Acceptance Criteria

- [] read he, glorot and fixup properly
- [] visualise the training dynamics
- [] show that there's a problem with the current random initialisation
- [] implement the expected initialisation strategy

**Comment by purzelrakete** [comment]
pytorch still doesn't have good default initialisations: https://github.com/pytorch/pytorch/issues/18182.

**Comment by purzelrakete** [comment]
## Papers

### Basics

- Understanding the difficulty of training deep feedforward neural networks [Glorot](https://proceedings.mlr.press/v9/glorot10a/glorot10a.pdf). This is about network attenuation in forward and backward passes, where the network is completely linear. When outputs are around zero, this is the case with tanh activations. 
- Delving Deep into Rectifiers: Surpassing Human-Level Performance on ImageNet Classification [He](https://arxiv.org/abs/1502.01852). Same as above but this time with Relu activations.
